internal/sirius/get_user_details_test.go

```diff
@@ -0,0 +1,110 @@
+package sirius
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/pact-foundation/pact-go/dsl"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetUserDetails(t *testing.T) {
+	t.Parallel()
+
+	pact := newPact()
+	defer pact.Teardown()
+
+	testCases := []struct {
+		name             string
+		setup            func()
+		expectedResponse User
+		expectedError    func(int) error
+	}{
+		{
+			name: "OK",
+			setup: func() {
+				pact.
+					AddInteraction().
+					Given("A donor exists").
+					UponReceiving("A request for the current user").
+					WithRequest(dsl.Request{
+						Method: http.MethodGet,
+						Path:   dsl.String("/lpa-api/v1/users/current"),
+					}).
+					WillRespondWith(dsl.Response{
+						Status: http.StatusOK,
+						Body: dsl.Like(map[string]interface{}{
+							"id":          dsl.Like(103),
+							"displayName": dsl.String("Test User"),
+							"roles":       dsl.Like([]string{"OPG User", "Case Manager"}),
+						}),
+						Headers: dsl.MapMatcher{"Content-Type": dsl.String("application/json")},
+					})
+			},
+			expectedResponse: User{
+				ID:          103,
+				DisplayName: "Test User",
+				Roles:       []string{"OPG User", "Case Manager"},
+			},
+		},
+		{
+			name: "OK",
+			setup: func() {
+				pact.
+					AddInteraction().
+					Given("A donor exists - logged in as the reduced fees user").
```

Comment:
I'd suggest changing this to:
```suggestion
					Given("I am a reduced fees user").
```

Then adding a matching block on the Sirius side telling it to log in as reduced.fees.user if it gets that setup state (just like the one for [Lay Team user](https://github.com/ministryofjustice/opg-sirius/blob/69cd997f4b053054f61c2706ab09c9e54ee98611/front-end/docker/pact-proxy/setup.go#L107-L110)).

I don't think you need the other Pact test above

Reply:
Updated 👍